src/main/java/mate/academy/spring/controller/CustomGlobalExceptionHandler.java

```diff
@@ -0,0 +1,63 @@
+package mate.academy.spring.controller;
+
+import java.time.LocalDateTime;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import mate.academy.spring.exception.DataProcessingException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+    @Override
+    protected ResponseEntity<Object> handleExceptionInternal(
+            Exception ex,
+            Object body,
+            HttpHeaders headers,
+            HttpStatus status,
+            WebRequest request) {
+        if (ex.getClass().equals(DataProcessingException.class)) {
+            Map<String, Object> responseBody = new LinkedHashMap<>();
+            responseBody.put("timestamp", LocalDateTime.now().toString());
+            responseBody.put("status", HttpStatus.INTERNAL_SERVER_ERROR);
+            responseBody.put("error", ex.getMessage());
+            return new ResponseEntity<>(responseBody, headers, status);
+        }
+        return super.handleExceptionInternal(ex, body, headers, status, request);
+    }
```

Comment:
@ExceptionHandler(DataProcessingException.class)

better add method with this annotations - this is how you work with your custom exceptions - if you want to process them in a different way

Reply:
I'll give it a try )